deploy/external-dns/pdns/patch/Deployment/external-dns.yaml

```diff
@@ -0,0 +1,15 @@
+- op: add
+  path: /spec/template/spec/containers/0/args/-
+  value: --provider=pdns
+- op: add
+  path: /spec/template/spec/containers/0/args/-
+  value: --pdns-server=http://10.43.0.11:8081
+- op: add
+  path: /spec/template/spec/containers/0/args/-
+  value: --pdns-api-key=abcdef123456
+- op: add
+  path: /spec/template/spec/containers/0/args/-
+  value: --domain-filter=metal.suse.network
```

Comment:
Is this still the correct domain . Since we are install pdns via helmchart and that has suse.baremetal should this include that?

Reply:
Those will be overwritten by the templates in template.d.

Reply:
ok I see is pulls in the env variable DNS_DOMAIN. but it seems you are not indicating that is this check


`    "pdns")
	CHECK_PDNS_SERVER_URL=$(grep 'pdns-server=' ${REPO_ROOT}/deploy/external-dns/${EXTERNAL_DNS_PROVIDER}/patch/Deployment/external-dns.yaml 2>/dev/null || true)
        if [ -z "${CHECK_PDNS_SERVER_URL}" ]
        then
            echo "ERROR: for PowerDNS external DNS provider, the following environment variables must be set before running render-templates.sh: POWERDNS_SERVER_URL, POWERDNS_API_TOKEN"
            exit 1
        fi
`

Reply:
This check is done, assuming after running scripts/render-templates.sh which is required.

Reply:
so will render templates error if DNS_DOMAIN is not set. Where in the docs does it mention  DNS_DOMAIN should be set and what it is used for?

Reply:
Good point! Yes env var DNS_DOMAIN is also required in order to render-templates.sh to work. Let me update.

Reply:
It might be helpful if we add an entry in the README.md or such explaining the values.   Also are we using these when we install the pdns helm chart?  Since the values.yaml for the pdns chart has the api key, url  and domain shouldn't we pull them from there? Also if we are installing this before pdns, are we concerned about external-dns starting before the pdns api is available?

Reply:
External-dns and it's providers are meant to be separated, and can be installed separately in any order. The only interaction between them are via REST APIs. They can start in any order and completely independent from each other. 
The reason we don't want to pull from `values.yaml` is that it is meant to contain the default values. They can be overwritten from helm command line. I'll add some text in README.md about the env vars.

Reply:
Thanks for adding an entry. While I agree external-dns and their providers are seperate, if you start external-dns before the provider it will get flooded with errors if it detects it needs to add a record and the provider has not yet been started.  so IMHO we should enforce the order just to be more "clean".

Reply:
Oh you mean deploy the powerdns helm chart prior to external_dns? That's a good idea. Though for CloudFlare, there's not much we can do.

Reply:
yes, exactly